### PR TITLE
feat(api): Implement Python Library API (fixes #7)

### LINKED ISSUE #7: [Phase 3] Task 3.1: Python Library API
## Description
Design and implement a clean, intuitive Python API for the library.

## Dependencies
- Task 2.3: Diff Output Formatters

## Acceptance Criteria
- Main API functions: `load_document()`, `validate_document()`, `diff_documents()`
- API is well-documented with docstrings
- API handles errors gracefully with custom exceptions
- API supports both file paths and in-memory objects
- API returns structured results (Pydantic models or dicts)
- API is intuitive for common use cases

## Tests
- Unit test: Load document via API
- Unit test: Validate document via API
- Unit test: Diff documents via API
- Unit test: API raises appropriate exceptions
- Unit test: API works with file paths
- Unit test: API works with file-like objects
- Integration test: Complete workflow via API

## Files to Create
- `src/yaml_diffs/api.py`
- `src/yaml_diffs/exceptions.py`
- `tests/test_api.py`
- `docs/api_reference.md`

## Phase
Phase 3: Interface Layers

### PR DESCRIPTION
**Pull request by noamoss**
## Summary

This PR implements a clean, intuitive Python Library API for yaml-diffs. It creates a unified `api.py` module that provides a cohesive interface to all core functionality, along with convenience functions for common workflows.

## Related Issues

Fixes #7

## Changes Made

- **Created `src/yaml_diffs/api.py`**: Unified API module with:
  - Re-exports of main functions (`load_document`, `validate_document`, `diff_documents`, `format_diff`)
  - Re-exports of exceptions and types
  - Three convenience functions:
    - `load_and_validate()` - Load and validate in one call
    - `diff_files()` - Diff two files directly
    - `diff_and_format()` - Diff and format in one call
  - Comprehensive Google-style docstrings with examples

- **Updated `src/yaml_diffs/__init__.py`**: 
  - Exports main API functions from `api.py`
  - Maintains full backward compatibility (direct imports from submodules still work)
  - Updated `__all__` to reflect new API structure

- **Created `tests/test_api.py`**: Comprehensive test suite with 33 tests covering:
  - Unit tests for all API functions
  - Error handling tests
  - Integration tests for complete workflows
  - Edge cases (Hebrew content, deeply nested documents, empty sections)
  - Tests with real example documents

- **Created `docs/api_reference.md`**: Complete API documentation with:
  - Overview and quick start guide
  - Function reference with examples
  - Common use cases
  - Error handling guide
  - Type documentation

## Testing

- [x] All 33 tests pass
- [x] Pre-commit hooks pass
- [x] Code follows project style guidelines
- [x] Type checking passes (mypy)
- [x] Linting passes (ruff)

## Acceptance Criteria

All acceptance criteria from Issue #7 are met:

- ✅ Main API functions: `load_document()`, `validate_document()`, `diff_documents()`
- ✅ API is well-documented with docstrings (Google-style with examples)
- ✅ API handles errors gracefully with custom exceptions
- ✅ API supports both file paths and in-memory objects (str, Path, TextIO)
- ✅ API returns structured results (Pydantic models or formatted strings)
- ✅ API is intuitive for common use cases (convenience functions provided)

## Checklist

- [x] Tests added/updated (33 comprehensive tests)
- [x] Documentation updated (API reference guide created)
- [x] Backward compatibility maintained
- [x] Code follows project conventions
- [x] All pre-commit hooks pass

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/yaml_diffs/api.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+from yaml_diffs.validator import validate_document
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+
+def load_and_validate(file_path: str | Path | TextIO) -> Document:
+    """Load and validate a document in one call.
+
+    This is a convenience function that combines `load_document()` and
```
Comment (noamoss) [inline src/yaml_diffs/api.py]: **Minor Documentation Issue**

The docstring says `load_and_validate()` "combines `load_document()` and `validate_document()`", but it actually just calls `validate_document()`. While technically correct (since `validate_document` already loads), the description could be clearer.

**Suggested fix:**
```python
def load_and_validate(file_path: str | Path | TextIO) -> Document:
    """Load and validate a document in one call.

    This is a convenience function that loads a YAML file and validates it against
    both the OpenSpec schema and Pydantic models. It is equivalent to calling
    `validate_document()`.

    ...
```

File: src/yaml_diffs/api.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    Raises:
+        YAMLLoadError: If the file cannot be read or parsed.
+        ImportError: If jsonschema is not available.
+        OpenSpecValidationError: If OpenSpec validation fails.
+        PydanticValidationError: If Pydantic validation fails.
+
+    Examples:
+        >>> doc = load_and_validate("examples/minimal_document.yaml")
+        >>> assert isinstance(doc, Document)
+        >>> assert doc.id == "law-1234"
```
Comment (noamoss) [inline src/yaml_diffs/api.py]: **Example Inconsistency**

The example uses `doc.id == "law-1234"`, but the actual minimal document in tests uses `"test-123"`. Consider updating the example to match the test fixtures, or use a more generic example.

**Suggested fix:**
```python
    Examples:
        >>> doc = load_and_validate("examples/minimal_document.yaml")
        >>> assert isinstance(doc, Document)
        >>> assert doc.id  # Document has an ID
```

**Comment by noamoss** [comment]
## 📋 Additional Review Feedback Addressed

I've applied the following changes based on the second code review:

### ✅ Changes Made:
- **Missing import fix**: Added `ValidationError` import to the error handling example in `docs/api_reference.md` (line 239)
- **Example consistency**: Fixed example ID from `"law-1234"` to `"test-123"` to match test fixtures (line 73)

### ✅ Validation:
- All 33 tests passed
- Code formatting verified
- Changes committed locally

### 📦 Next Step:
Pushing changes to remote branch...

---
Ready for re-review! 🚀

File: docs/api_reference.md [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+
+### 2. Validate a Document
+
+```python
+from yaml_diffs import validate_document, ValidationError
+
+try:
+    doc = validate_document("document.yaml")
+    print("Document is valid!")
+except ValidationError as e:
```
Comment (noamoss) [inline docs/api_reference.md]: **Missing Import in Documentation Example**

The example code uses `ValidationError` but doesn't show the import statement. This could confuse users who try to copy-paste the example.

**Suggested fix:**
```python
from yaml_diffs import validate_document, ValidationError

try:
    doc = validate_document("document.yaml")
    print("Document is valid!")
except ValidationError as e:
    print(f"Validation failed: {e}")
```

Note: This is already noted in an existing review comment, but I wanted to emphasize it's important for user experience.
